pkg/cmd/openshift-controller-manager/cmd.go

```diff
@@ -2,119 +2,39 @@ package openshift_controller_manager
 
 import (
 	"context"
-	"errors"
-	"fmt"
-	"io"
-	"io/ioutil"
 	"os"
-	"path"
 
 	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/utils/clock"
 
-	kerrors "k8s.io/apimachinery/pkg/api/errors"
-	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/runtime/serializer"
-	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
-	"k8s.io/client-go/tools/clientcmd/api"
-	"k8s.io/klog/v2"
-	kcmdutil "k8s.io/kubectl/pkg/cmd/util"
-	"k8s.io/kubectl/pkg/util/templates"
-
-	configv1 "github.com/openshift/api/config/v1"
-	openshiftcontrolplanev1 "github.com/openshift/api/openshiftcontrolplane/v1"
-	"github.com/openshift/library-go/pkg/config/helpers"
-	"github.com/openshift/library-go/pkg/serviceability"
+	"github.com/openshift/library-go/pkg/controller/controllercmd"
+	"github.com/openshift/openshift-controller-manager/pkg/version"
 )
 
-type OpenShiftControllerManager struct {
-	ConfigFilePath string
-	Output         io.Writer
-}
-
-var longDescription = templates.LongDesc(`
-	Start the OpenShift controllers`)
-
-func NewOpenShiftControllerManagerCommand(name string, out, errout io.Writer, ctx context.Context) *cobra.Command {
-	options := &OpenShiftControllerManager{Output: out}
-
-	cmd := &cobra.Command{
-		Use:   name,
-		Short: "Start the OpenShift controllers",
-		Long:  longDescription,
-		Run: func(c *cobra.Command, args []string) {
-			kcmdutil.CheckErr(options.Validate())
-
-			serviceability.StartProfiler()
-
-			if err := options.StartControllerManager(ctx); err != nil {
-				if kerrors.IsInvalid(err) {
-					if details := err.(*kerrors.StatusError).ErrStatus.Details; details != nil {
-						fmt.Fprintf(errout, "Invalid %s %s\n", details.Kind, details.Name)
-						for _, cause := range details.Causes {
-							fmt.Fprintf(errout, "  %s: %s\n", cause.Field, cause.Message)
-						}
-						os.Exit(255)
-					}
-				}
-				klog.Fatal(fmt.Sprintf("fatal error occurred: %v", err))
-			}
-		},
-	}
+const (
+	podNameEnv      = "POD_NAME"
+	podNamespaceEnv = "POD_NAMESPACE"
+)
 
-	flags := cmd.Flags()
-	// This command only supports reading from config
-	flags.StringVar(&options.ConfigFilePath, "config", options.ConfigFilePath, "Location of the master configuration file to run from.")
-	cmd.MarkFlagFilename("config", "yaml", "yml")
+func NewOpenShiftControllerManagerCommand(name string) *cobra.Command {
+	cm := &OpenShiftControllerManager{}
+	c := controllercmd.NewControllerCommandConfig("openshift-controller-manager", version.Get(), cm.Run, clock.RealClock{}).
+		WithComponentOwnerReference(&corev1.ObjectReference{
+			Kind:      "Pod",
+			Name:      os.Getenv(podNameEnv),
+			Namespace: os.Getenv(podNamespaceEnv),
+		})
+
+	// We need to set a custom lock name since that's what Ansible used to set.
+	c.ResourceLockName = "openshift-master-controllers"
```

Comment:
This is the proposed field to be added to library-go to allow us to set custom resource lock name.